Assets/Scripts/SS3D/Systems/Interactions/InteractionController.cs

```diff
@@ -38,32 +41,58 @@ public override void OnStartClient()
 
             _radialView = SystemLocator.Get<RadialInteractionView>();
             _camera = SystemLocator.Get<CameraSystem>().PlayerCamera.GetComponent<Camera>();
-            Controls controls = SystemLocator.Get<InputSystem>().Inputs;
-            _otherControls = controls.Other;
+            _inputSystem = SystemLocator.Get<InputSystem>();
+            Controls controls = _inputSystem.Inputs;
+            _controls = controls.Interactions;
             _hotkeysControls = controls.Hotkeys;
-            _otherControls.PrimaryClick.performed += HandlePrimaryClick;
-            _otherControls.SecondaryClick.performed += HandleSecondaryClick;
+            _controls.PerformInteraction.performed += HandlePerform;
+            _controls.ViewInteractions.performed += HandleView;
             _hotkeysControls.Use.performed += HandleUse;
+            _inputSystem.ToggleActionMap(_controls, true);
         }
 
         public override void OnStopClient()
         {
             base.OnStopClient();
             
-            _otherControls.PrimaryClick.performed -= HandlePrimaryClick;
-            _otherControls.SecondaryClick.performed -= HandleSecondaryClick;
+            _controls.PerformInteraction.performed -= HandlePerform;
+            _controls.ViewInteractions.performed -= HandleView;
             _hotkeysControls.Use.performed -= HandleUse;
+            _inputSystem.ToggleActionMap(_controls, false);
         }
 
+        /// <summary>
+        /// Runs the most prioritised action
+        /// </summary>
         [Client]
-        private void HandlePrimaryClick(InputAction.CallbackContext callbackContext)
+        private void HandlePerform(InputAction.CallbackContext callbackContext)
         {
-            RunPrimaryInteraction();
```

Comment:
I like the RunPrimaryInteraction, we just have to read one line to understand what's happening and it's just more modular if we want to modify it later on. Could you add it back ?